### ISSUE TITLE
[BUG] Cannot import pysindy into Google Colab due to version conflict issues

### ISSUE THREAD

**Issue by jdawg86** [dynamicslab/pysindy#174]
I cannot seem to import the pysindy library at all in Google colab due to some version conflict issues. The install of pysindy seems to work well enough and beforehand I do have deprecation issues with installing scikit-learn but that does not seem to affect pysindy. It seems that there is a version conflict between the colab environment and pysindy.

### Reproducing code example:

```python
!pip install scikit-learn[alldeps]==0.19.1
!pip install pysindy

import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
from matplotlib.cm import rainbow
import numpy as np
from scipy.integrate import odeint
from scipy.io import loadmat

import pysindy as ps
```

### Error message:
```
ContextualVersionConflict                 Traceback (most recent call last)
[<ipython-input-8-8d2c0c79f4b0>](https://localhost:8080/#) in <module>()
      9 #from scipy.io import loadmat
     10 
---> 11 import pysindy as ps
     12 
     13 

4 frames
[/usr/local/lib/python3.7/dist-packages/pysindy/__init__.py](https://localhost:8080/#) in <module>()
      3 
      4 try:
----> 5     __version__ = get_distribution(__name__).version
      6 except DistributionNotFound:
      7     pass

[/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py](https://localhost:8080/#) in get_distribution(dist)
    464         dist = Requirement.parse(dist)
    465     if isinstance(dist, Requirement):
--> 466         dist = get_provider(dist)
    467     if not isinstance(dist, Distribution):
    468         raise TypeError("Expected string, Requirement, or Distribution", dist)

[/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py](https://localhost:8080/#) in get_provider(moduleOrReq)
    340     """Return an IResourceProvider for the named module or requirement"""
    341     if isinstance(moduleOrReq, Requirement):
--> 342         return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
    343     try:
    344         module = sys.modules[moduleOrReq]

[/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py](https://localhost:8080/#) in require(self, *requirements)
    884         included, even if they were already activated in this working set.
    885         """
--> 886         needed = self.resolve(parse_requirements(requirements))
    887 
    888         for dist in needed:

[/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py](https://localhost:8080/#) in resolve(self, requirements, env, installer, replace_conflicting, extras)
    775                 # Oops, the "best" so far conflicts with a dependency
    776                 dependent_req = required_by[req]
--> 777                 raise VersionConflict(dist, req).with_context(dependent_req)
    778 
    779             # push the new requirements onto the stack

ContextualVersionConflict: (pytest 3.6.4 (/usr/local/lib/python3.7/dist-packages), Requirement.parse('pytest<6.0.0,>=5.4.3'), {'derivative'})
```

**Comment by Jacob-Stevens-Haas** [comment]
I'm not sure why colab needs pytest 3.6.4, but this is an issue in [`derivative`](https://github.com/andgoldschmidt/derivative), not `pysindy` (related people work on both).  

The error was that [pytest was listed as a dependency](https://github.com/andgoldschmidt/derivative/blob/14593c45f574ff3a7bcc2f97939f4d41abb8e535/pyproject.toml#L26), when it was really a developer dependency.  I think that has been fixed in the current version of the repo, but the master branch needs to be republished to pypi. (Building master locally removes pytest from dist-info/METADATA)

**Comment by andgoldschmidt** [comment]
Should be fixed in derivative 0.4.2

I've moved pytest from a `[dev-dependency]` to an `[extras]` called dev. This means `poetry add -D` doesn't really apply anymore, but that's not really a big price to pay in order to add a very explicit pytest dependency for derivative. 

[derivative #a35b0ce](https://github.com/andgoldschmidt/derivative/commit/a35b0cefc1be0a9e5c6513ee100bb4575edd0f30)
